src/test/kotlin/com/gamingbacklog/api/gamingbacklogapi/unit/controllers/LibraryControllerTests.kt

```diff
@@ -0,0 +1,173 @@
+      libraries.add(library2)
+      given(libraryService.getAll()).willReturn(libraries)
+      requestBuilder.runGetRequest(endpoint)
+        .andExpect(status().isOk)
+        .andExpect(jsonPath("$[0].name", equalTo("Backlog")))
+        .andExpect(jsonPath("$[0].games", equalToObject(ArrayList<String>())))
+        .andExpect(jsonPath("$[0].id", equalTo("id1")))
+        .andExpect(jsonPath("$[1].name", equalTo("Owned Games")))
+        .andExpect(jsonPath("$[1].games[0]", equalTo("gameId1")))
+        .andExpect(jsonPath("$[1].games[1]", equalTo("gameId2")))
+        .andExpect(jsonPath("$[1].id", equalTo("id2")))
+    }
+  }
+
+  @Nested
+  @DisplayName("Tests for getSingleLibrary")
+  inner class GetSingleLibrary {
+    // returns 200
+    @Test
+    fun shouldReturnLibrarySuccessfully() {
+      val library = Library(id1, "Owned Games", ArrayList())
+      library.games.add("gameId1")
+      library.games.add("gameId2")
+      given(libraryService.getSingle(any())).willReturn(library)
+      endpoint += "$id1/"
+      requestBuilder.runGetRequest(endpoint)
+        .andExpect(status().isOk)
+        .andExpect(jsonPath("$.name", equalTo("Owned Games")))
+        .andExpect(jsonPath("$.games[0]", equalTo("gameId1")))
+        .andExpect(jsonPath("$.games[1]", equalTo("gameId2")))
+        .andExpect(jsonPath("$.id", equalTo(id1)))
+    }
+    // returns 404
+    @Test
+    fun shouldReturnNoLibrary() {
+      given(libraryService.getSingle(any())).willReturn(null)
+      endpoint += "$id1/"
+      requestBuilder.runGetRequest(endpoint)
+        .andExpect(status().isOk)
+        .andExpect(jsonPath("$").doesNotExist())
+    }
+  }
+
+  @Nested
+  @DisplayName("Tests for createLibrary")
+  inner class CreateLibrary {
+    // creates library successfully
+    @Test
+    fun shouldSuccessfullyCreateLibraryWithNoGamesPassedIn() {
+      val library = Library(id2, "Backlog", ArrayList())
+      given(libraryService.create(any())).willReturn(library)
+      val libraryRequest = LibraryRequest("Backlog", null)
+      requestBuilder.runPostRequest(endpoint, requestToString(libraryRequest))
+        .andExpect(status().isCreated)
+        .andExpect(jsonPath("$.name", equalTo(library.name)))
+        .andExpect(jsonPath("$.games", equalTo(library.games)))
+    }
+
+    @Test
+    fun shouldSuccessfullyCreateLibraryWithGamesPassedIn() {
+      val library = Library(id2, "Backlog", ArrayList())
+      library.games.add(id1)
+      given(libraryService.create(any())).willReturn(library)
+      val libraryRequest = LibraryRequest("Backlog", arrayListOf(id1))
+      requestBuilder.runPostRequest(endpoint, requestToString(libraryRequest))
+        .andExpect(status().isCreated)
+        .andExpect(jsonPath("$.name", equalTo(library.name)))
+        .andExpect(jsonPath("$.games[0]", equalTo(id1)))
+    }
+  }
+
+  @Nested
+  @DisplayName("Tests for addToLibrary")
+  // TODO: update tests when master library functionality is complete
```

Comment:
Do we have a ticket for this? Might be good to note that in the TODO itself just so we don't lose track

Reply:
added ticket in there